app/screens/channel_peek/index.js

```diff
@@ -1,37 +1,60 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {bindActionCreators} from 'redux';
-import {connect} from 'react-redux';
+import {realmConnect} from 'realm-react-redux';
 
-import {getPostIdsInChannel} from 'mattermost-redux/selectors/entities/posts';
-import {getMyChannelMember} from 'mattermost-redux/selectors/entities/channels';
-import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
-import {loadPostsIfNecessaryWithRetry, markChannelViewedAndRead} from 'app/actions/views/channel';
-import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
+import {General, Preferences} from 'app/constants';
+
+import {markChannelsViewedAndRead} from 'app/realm/actions/channel';
+import {loadPostsWithRetry} from 'app/realm/actions/post';
+import {getPostListIdsWithSeparators} from 'app/realm/selectors/post';
+import {getTheme} from 'app/realm/selectors/preference';
+import options from 'app/store/realm_options';
 
 import ChannelPeek from './channel_peek';
 
-function mapStateToProps(state, ownProps) {
-    const channelId = ownProps.channelId;
-    const myMember = getMyChannelMember(state, channelId);
+function mapPropsToQueries(realm, ownProps) {
+    const {channelId, currentUserId} = ownProps;
+    const general = realm.objectForPrimaryKey('General', General.REALM_SCHEMA_ID) || General.REALM_EMPTY_OBJECT;
+    const themePreferences = realm.objects('Preference').filtered(`category="${Preferences.CATEGORY_THEME}"`);
 
-    return {
-        channelId,
-        currentUserId: getCurrentUserId(state),
-        postIds: getPostIdsInChannel(state, channelId),
-        lastViewedAt: myMember && myMember.last_viewed_at,
-        theme: getTheme(state),
-    };
+    const currentUser = realm.objectForPrimaryKey('User', general?.currentUserId || '') || General.REALM_EMPTY_OBJECT;
```

Comment:
No need for the or in the query and at this point you already have the currentUserId, so no need to access it from general

Reply:
Yes, missed that